proxy/request_test.go

```diff
@@ -0,0 +1,25 @@
+package proxy
+
+import (
+	"testing"
+)
+
+func TestBadMessage(t *testing.T) {
+	req := ""
+	resp := handleRequest([]byte(req))
+
+	expected := `{"id":null,"error":{"errcode":"M_NOT_JSON","error":"unexpected end of JSON input"}}`
```

Comment:
ID is null because that's what JSON-RPC specifies in this instance. I went with making it look like JSON-RPC except where there seemed good reasons to do otherwise.